### PR TITLE
[Merged by Bors] - Hare metric

### PR DESCRIPTION
**Pull request by gavraz**
## Motivation
Better metrics for hare.

## Changes
Count number of running consensus processes.
Fix attempt to calc active set size on consensus view if not synced.

### CODE REVIEW

**Review by barakshani — APPROVED** [review]

File: hare/hare.go [hunk 1]
```diff
@@ -206,6 +206,9 @@ func (h *Hare) onTick(id types.LayerID) {
 		return
 	}
 
+	// call to start the calculation of active set size beforehand
+	go h.rolacle.IsIdentityActiveOnConsensusView(h.nid.Key, id)
```
Comment (y0sher) [inline hare/hare.go:210]: what does this has to do with metrics ?

File: hare/hare.go [hunk 2]
```diff
@@ -5,6 +5,7 @@ import (
 	"github.com/spacemeshos/go-spacemesh/common/types"
 	"github.com/spacemeshos/go-spacemesh/hare/config"
 	"github.com/spacemeshos/go-spacemesh/log"
+	"go.uber.org/atomic"
```
Comment (y0sher) [inline hare/hare.go]: why not use the go stdlib atomic ?

File: miner/block_builder.go [hunk 3]
```diff
@@ -402,6 +402,7 @@ func (t *BlockBuilder) acceptBlockData() {
 		case layerID := <-t.beginRoundEvent:
 			if !t.syncer.IsSynced() {
 				t.Debug("builder got layer %v not synced yet", layerID)
+				continue
```
Comment (y0sher) [inline miner/block_builder.go:405]: what does this has to do with metrics ?

**Review by y0sher — CHANGES_REQUESTED** [review]
please see my comments

File: miner/block_builder.go (hunk 3 above)
Comment (gavraz) [reply miner/block_builder.go:405]: Shaatnez shel Barak

File: hare/hare.go (hunk 1 above)
Comment (gavraz) [reply hare/hare.go:210]: shaatnez shel barak

File: hare/hare.go (hunk 2 above)
Comment (gavraz) [reply hare/hare.go]: Changed to golangs atomic
please review the usage

**Review by y0sher — APPROVED** [review]

File: miner/block_builder.go (hunk 3 above)
Comment (y0sher) [reply miner/block_builder.go:405]: I'd still rather you put it in a different pr

File: hare/hare.go (hunk 1 above)
Comment (y0sher) [reply hare/hare.go:210]: I'd still rather you put it in a different pr

**Comment by gavraz** [comment]
bors merge

**Pull request closed without merging**